From 93717c663c2157e0c7930c1ef1469435d899342a Mon Sep 17 00:00:00 2001
From: Chris Bieneman <beanz@apple.com>
Date: Mon, 24 Jul 2017 13:41:08 -0700
Subject: [PATCH 08/54] Rename SWIFT_TABLEGEN to LLVM_TABLEGEN

Swift doesn't produce its own tablegen and by faking out the LLVM CMake infrastructure into thinking it does all of LLVM's build cross compilation support breaks.
---
 cmake/modules/AddSwiftTableGen.cmake       | 2 +-
 cmake/modules/SwiftSharedCMakeConfig.cmake | 5 ++---
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/cmake/modules/AddSwiftTableGen.cmake b/cmake/modules/AddSwiftTableGen.cmake
index d8d80f07d15..4b7e3e0a8d3 100644
--- a/cmake/modules/AddSwiftTableGen.cmake
+++ b/cmake/modules/AddSwiftTableGen.cmake
@@ -3,7 +3,7 @@ include(TableGen)
 # This needs to be a macro since tablegen (which is a function) needs to set
 # variables in its parent scope.
 macro(swift_tablegen)
-  tablegen(SWIFT ${ARGN})
+  tablegen(LLVM ${ARGN})
 endmacro()
 
 # This needs to be a macro since add_public_tablegen_target (which is a
diff --git a/cmake/modules/SwiftSharedCMakeConfig.cmake b/cmake/modules/SwiftSharedCMakeConfig.cmake
index 1c38d21ffb5..2dc78ba0e29 100644
--- a/cmake/modules/SwiftSharedCMakeConfig.cmake
+++ b/cmake/modules/SwiftSharedCMakeConfig.cmake
@@ -68,9 +68,9 @@ macro(swift_common_standalone_build_config_llvm product is_cross_compiling)
     set(${product}_NATIVE_LLVM_TOOLS_PATH "${LLVM_TOOLS_BINARY_DIR}")
   endif()
 
-  find_program(SWIFT_TABLEGEN_EXE "llvm-tblgen" "${${product}_NATIVE_LLVM_TOOLS_PATH}"
+  find_program(LLVM_TABLEGEN_EXE "llvm-tblgen" "${${product}_NATIVE_LLVM_TOOLS_PATH}"
     NO_DEFAULT_PATH)
-  if ("${SWIFT_TABLEGEN_EXE}" STREQUAL "SWIFT_TABLEGEN_EXE-NOTFOUND")
+  if ("${LLVM_TABLEGEN_EXE}" STREQUAL "LLVM_TABLEGEN_EXE-NOTFOUND")
     message(FATAL_ERROR "Failed to find tablegen in ${${product}_NATIVE_LLVM_TOOLS_PATH}")
   endif()
 
@@ -219,8 +219,6 @@ macro(swift_common_unified_build_config product)
   set(${product}_NATIVE_LLVM_TOOLS_PATH "${CMAKE_BINARY_DIR}/bin")
   set(${product}_NATIVE_CLANG_TOOLS_PATH "${CMAKE_BINARY_DIR}/bin")
   set(LLVM_PACKAGE_VERSION ${PACKAGE_VERSION})
-  set(SWIFT_TABLEGEN_EXE llvm-tblgen)
-  set(SWIFT_TABLEGEN_TARGET "${SWIFT_TABLEGEN_EXE}")
   set(LLVM_CMAKE_DIR "${CMAKE_SOURCE_DIR}/cmake/modules")
 
   # If cmark was checked out into tools/cmark, expect to build it as
